### DISCUSSION TITLE
Limiting Queryset

### DISCUSSION THREAD

**Discussion by ByteOven** [carltongibson/django-filter#1355] [Q&A] [answered]
Was wondering is there a option to limit the queryset.

When I try to add limit / offset, the QS gives an error :
`Cannot filter a query once a slice has been taken. `

**Comment by carltongibson** [comment]
Filter before limiting (with the slice)

**Reply by ByteOven** [reply]
Would you mind providing a working demo ?

**Comment by saadmk11** [comment, marked as answer]
If you are using `FilterView`, `BaseFilterView`, or any similar approach you can use limit after the filtering is done by `django-filter`.
For Example,

```python
def get(self, request, *args, **kwargs):
    filterset_class = self.get_filterset_class()
    self.filterset = self.get_filterset(filterset_class)
    
    if not self.filterset.is_bound or self.filterset.is_valid() or not self.get_strict():
        self.object_list = self.filterset.qs
    else:
        self.object_list = self.filterset.queryset.none()
    
    context = self.get_context_data(
        filter=self.filterset,
        object_list=self.object_list[:10] # <-------------- Limit Here
    )
    return self.render_to_response(context)
```

Or you can override the `filter_queryset()` method of your filterset class.
For Example,

```python
def filter_queryset(self, queryset):
    return super().filter_queryset(queryset)[:10]
```

**Reply by carltongibson** [reply]
Thanks for your contributions @saadmk11! 🏅

**Reply by saadmk11** [reply]
@carltongibson Was looking through the issue tracker to contribute back to `django-filter` which I've been using for a long time, Glad I was able to help! :)
